Makefile

```diff
@@ -133,8 +134,11 @@ test: all
 ############# LOCAL INSTALLATION AND DOWNSTREAM PACKAGING ##############
 #
 version: VERSION mods/ra/mod.yaml mods/cnc/mod.yaml mods/d2k/mod.yaml mods/ts/mod.yaml mods/modcontent/mod.yaml mods/all/mod.yaml
-	@sh -c '. ./packaging/functions.sh; set_engine_version $(VERSION) .'
-	@sh -c '. ./packaging/functions.sh; set_mod_version $(VERSION) mods/ra/mod.yaml mods/cnc/mod.yaml mods/d2k/mod.yaml mods/ts/mod.yaml mods/modcontent/mod.yaml mods/all/mod.yaml'
+ifeq ($(GIT_VERSION), git-)
+	$(error Unable to determine version from git repository)
```

Comment:
Does this mean you are now requiring git even though a release tarball was downloaded?

Reply:
I believe the motivation here is to make it so that `make version` is unnecessary when using the release tarball. This is a good idea, provided we don't accidentally break other cases (e.g. your comment about setting VERSION).

Reply:
> I believe the motivation here is to make it so that `make version` is unnecessary when using the release tarball.

`make version` was not necessary for release tarballs and didn't work. The only difference from this PR is that it outputs an error instead of setting the version to `git-`.

> This is a good idea, provided we don't accidentally break other cases (e.g. allow version overrides by explicitly calling `make version VERSION=foo`)

The example use case did not work before either, because you can't override the variable `VERSION`.

See https://github.com/OpenRA/OpenRA/pull/19156#discussion_r579630574

Reply:
It has always worked for me, and as part of our packaging scripts!

Reply:
Yeah, I noticed my mistake.